dirsrvtests/tests/suites/acl/aci_excl_filter_test.py

```diff
@@ -0,0 +1,146 @@
+    ou.create(properties={'ou': name})
+
+
+def add_user_entry(topo, user, name, pw, myparent):
+    dn = 'ou=%s,%s' % (name, myparent)
+    properties = {
+            'uid': name,
+            'cn' : name,
+            'sn' : name,
+            'uidNumber': '1000',
+            'gidNumber': '2000',
+            'homeDirectory':'/home/{}'.format(name),
+            'telephonenumber': '+1 222 333-4444',
+            'userpassword' : pw,
+        }
+
+    user.create(properties=properties)
+    return user
+
+
+def test_aci_with_exclude_filter(topo):
+    """
+       Test an ACI(Access control instruction) which contains an extensible filter.
+       Test that during the schema reload task there is a small window where the new schema is not loaded
+       into the asi hashtables - this results in searches not returning entries.
+    :id: test_aci_with_exclude_filter
+    :setup: Standalone instance
+    :steps:
+        1. Bind to a new Standalone instance
+        2. Generate text for the Access Control Instruction(ACI) and add to the standalone instance
+           -Create a test user 'admin' with a marker -> deniedattr = 'telephonenumber'
+        3. Create 2 top Organizational units (ou) under the same root suffix
+        4. Create 2 test users for each Organizational unit (ou) above with the same username 'admin'
+        5. Bind to the Standalone instance as the user 'admin' from the ou created in step 4 above
+           - Search for user(s) ' admin in the subtree that satisfy this criteria:
+               DEFAULT_SUFFIX, ldap.SCOPE_SUBTREE, cn_filter, [deniedattr, 'dn'] 
+        6.  The search should return 2 entries with the username 'admin'
+        7.  Verify that the users found do not have the --> deniedattr = 'telephonenumber' marker
+    :expectedresults:
+        1. Operation should be successful
+        2. Operation should be successful
+        3. Operation should be successful
+        4. PASS - users found do not have the --> deniedattr = 'telephonenumber' marker
+
+    """
+    log.info('Bind as root DN')
+  
+    log.info('Create an OU for them')    
+    ous = OrganizationalUnit(topo.standalone, DEFAULT_SUFFIX)
+#   # Create the OU for them
+  
+    log.info('Create an top org users')
+    users = UserAccounts(topo.standalone, DEFAULT_SUFFIX)
+    log.info('Add aci which contains extensible filter.')
+    ouname = 'outest'
+    username = 'admin'
+    passwd = 'Password'
+    deniedattr = 'telephonenumber'
+    log.info('Add aci which contains extensible filter.')
+
+    aci_text = ('(targetattr = "{}")'.format(deniedattr) +
+                 '(target = "ldap:///{}")'.format(DEFAULT_SUFFIX) +
+                 '(version 3.0;acl "admin-tel-matching-rule-outest";deny (all)' +
+                 '(userdn = "ldap:///{}??sub?(&(cn={})(ou:dn:={}))");)'.format(DEFAULT_SUFFIX, username, ouname))
+    try:
+         topo.standalone.modify_s(DEFAULT_SUFFIX, [(ldap.MOD_ADD, 'aci', ensure_bytes(aci_text))])
```

Comment:
I think you can use Domain to add aci instead of modify_s:

`from lib389.idm.domain import Domain`
`Domain(topo.standalone, DEFAULT_SUFFIX).add("aci", aci_text)`

There is usage in suites/acl/acivattr_test.py

But I am working with aci for the first time so I would like someone else to confirm.
@vashirov Can I ask you to also review, please? Thanks!